flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/stream/sql/SqlITCase.scala

```diff
@@ -469,6 +473,148 @@ class SqlITCase extends StreamingWithStateTestBase {
     assertEquals(expected.sorted, StreamITCase.testResults.sorted)
   }
 
+  @Test
+  def testArrayElementAtFromTableForTuple(): Unit = {
+
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env)
+    StreamITCase.clear
+
+    val data = List(
+      (1, Array((12, 45), (2, 5))),
```

Comment:
1. Add null tuple input test
2. Add nested tuple input test

Reply:
Added null check.
for nested tuple input, it wont work for now due to https://issues.apache.org/jira/browse/CALCITE-2162. I've submitted a fix to it, should be available in Calcite 1.16.